src/Flow.cpp

```diff
@@ -16,4 +17,15 @@ Flow::Flow(const std::vector<std::shared_ptr<NetworkDevice>>& path,
     this->path = path;
     this->distances_ns = distances_ns;
     id = Flow::last_given_flow_id++;
+}
+
+void Flow::Send() {
+    std::shared_ptr<NetworkDevice> sender = path.front();
+    std::shared_ptr<NetworkDevice> receiver = path.back();
```

Comment:
After the suggested changes in https://github.com/Cloud-Storage-Team/algnet/pull/17 this will not be needed anymore.